sqlglot/lineage.py

```diff
@@ -276,8 +276,81 @@ def to_node(
                 reference_node_name=reference_node_name,
                 trim_selects=trim_selects,
             )
+
+        elif scope.pivots:
+            pivot = next(
+                (p for p in scope.pivots if p.alias_or_name == c.table and not p.unpivot), None
+            )
+            if pivot and isinstance(pivot.args.get("columns"), list):
```

Comment:
If this branch is not taken, we won't get a placeholder for this column.

Reply:
I precalculate the possible pivot before entering the if cases. With that the placeholder branch should be hit, if the there is no source found and there is not pivot